address_lookup_tables/src/address_lookup_table_store.rs

```diff
@@ -0,0 +1,218 @@
+                                if let Some(acc) = acc {
+                                    this.save_account(&batch[index], &acc.data);
+                                }
+                            }
+                        }
+                        Err(e) => {
+                            log::error!(
+                                "error loading {} alts with error {}",
+                                batch.len(),
+                                e.to_string()
+                            );
+                        }
+                    };
+                })
+            });
+            if tokio::time::timeout(Duration::from_secs(60), futures::future::join_all(tasks))
+                .await
+                .is_err()
+            {
+                log::error!("timeout loading {} alts", alts_list.len());
+            }
+        }
+        LRPC_ALTS_IN_STORE.set(self.map.len() as i64);
+    }
+
+    pub fn save_account(&self, address: &Pubkey, data: &[u8]) {
+        let lookup_table = AddressLookupTable::deserialize(data).unwrap();
+        if self
+            .map
+            .insert(*address, lookup_table.addresses.to_vec())
+            .is_none()
+        {
+            LRPC_ALTS_IN_STORE.inc();
+        }
+        drop(lookup_table);
+    }
+
+    pub async fn reload_if_necessary(
+        &self,
+        alt_messages: &[&solana_sdk::message::v0::MessageAddressTableLookup],
+    ) {
+        let accounts_to_load = alt_messages
+            .iter()
+            .filter_map(|alt| match self.map.get(&alt.account_key) {
+                Some(alt_data) => {
+                    let size = alt_data.len() as u8;
+                    if alt.readonly_indexes.iter().any(|x| *x >= size)
+                        || alt.writable_indexes.iter().any(|x| *x >= size)
+                    {
+                        Some(alt.account_key)
+                    } else {
+                        None
+                    }
+                }
+                None => Some(alt.account_key),
+            })
+            .collect_vec();
+        self.load_alts_list(&accounts_to_load).await;
+    }
+
+    pub async fn reload_alt_account(&self, address: &Pubkey) {
+        log::info!("Reloading {address:?}");
+
+        let account = match self
+            .rpc_client
+            .get_account_with_commitment(address, CommitmentConfig::processed())
+            .await
+        {
+            Ok(acc) => acc.value,
+            Err(e) => {
+                log::error!(
+                    "Error for fetching address lookup table {} error :{}",
+                    address.to_string(),
+                    e.to_string()
+                );
+                None
+            }
+        };
+        match account {
+            Some(account) => {
+                self.save_account(address, &account.data);
+            }
+            None => {
+                log::error!("Cannot find address lookup table {}", address.to_string());
```

Comment:
pls log what you do here  - i.e. "skip reload of account foo and continue"
would use _warn_ level

Reply:
Let it be error for now, to see how often it occurs and then could be changed to warn later when it is too verbose